py/runtime.c

```diff
@@ -110,6 +110,13 @@ void mp_init(void) {
     }
     #endif
 
+    #if MICROPY_PY_SYS_BUILTIN_MODULE_NAMES
+    extern mp_obj_tuple_t mp_sys_builtin_module_names_obj; // see py/objmodule.c
+    for (size_t i = 0; i < mp_sys_builtin_module_names_obj.len; ++i) {
```

Comment:
perhaps `__main__` ought to be cut out here; Some CPython implementations show `__main__` in their `sys.builtin_module_names`, and some don't.

I think it's okay to retain it.